### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by mirouhml**
# Project requirements:
- Create class Person with the following:
  - Instance vars: @id, @name, and @age.
  - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
  - Getters for @id, @name, and @age.
  - Setters for @name and @age.
  - Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
  - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- Create class Student with the following:
  - Inherits from Person.
  - Constructor extends parent's constructor by adding @classroom and a parameter for it.
  - Method play_hooky that returns "¯\(ツ)/¯".
- Create class Teacher with the following:
  - Inherits from Person.
  - Constructor extends parent's constructor by adding @specialization and a parameter for it.
  - Override can_use_services? so it always returns true.
- Each class should be saved in a separate file.

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+
+class Teacher < Person
+  def initialize(age, specialization, name = 'unknown')
```
Comment (micheaol) [inline teacher.rb]: Please ensure that you add `parent_permission = true` to the `initialize params` according to the project requirement.

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @mirouhml , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: teacher.rb (hunk 1 above)
Comment (mirouhml) [reply teacher.rb]: Alright I will do it.

**Review by elyor-doniyorov — APPROVED** [review]
Hi @mirouhml, 

In my opinion, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by mirouhml** [comment]
@elyor-doniyorov @micheaol  thank you both for the reviews!

**Pull request merged**